pkgs/misc/emulators/yuzu-mainline/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchgit, cmake, libressl, wrapQtAppsHook, SDL2, qtbase, python2 }:
+
+stdenv.mkDerivation {
+  pname = "yuzu-mainline";
+  version = "unstable-2020-04-13";
+
+  src = fetchgit {
+    url = "https://github.com/yuzu-emu/yuzu-mainline";
+    rev = "20c42dc9ea8bdc208d9d274350af2ff347876c76";
+    sha256 = "0s6vnvbyshbrl1xxiwdrxc2jl75jyjlf4kidl88c7pwj3m2gcwss";
```

Comment:
I'm getting an error with this sha256:
```
hash mismatch in fixed-output derivation '/nix/store/g3m9wqmhsp7ffvkldqkgc7fl57f92xxg-yuzu-mainline-20c42dc':
  wanted: sha256:0s6vnvbyshbrl1xxiwdrxc2jl75jyjlf4kidl88c7pwj3m2gcwss
  got:    sha256:0486nhpvl824ajksn496g0rfi9hismm8xzwdka5lplxrdmk961q2
```

Reply:
I think the options used are causing git to be non-deterministic, hence the changing hashes.

Specifically `leaveDotGit` , this seems to be related: https://github.com/NixOS/nixpkgs/issues/8567

Reply:
Oh, duh. The previous attempt at packaging this application had the same problem: https://github.com/NixOS/nixpkgs/pull/58312

Reply:
I've crafted my own version of yuzu-mainline in my NUR repo: https://github.com/JoshuaFern/nur-package-lab/blob/master/pkgs/misc/emulators/yuzu-mainline/default.nix

I believe this version avoids the hashing issue.

Reply:
At the moment Animal Crossing support is [a tad weird](https://www.patreon.com/posts/36005316), that's not the fault of this. I've tested Super Mario Odyssey and various [example homebrew applications](https://github.com/switchbrew/switch-examples) and they all seem to work well. 

As for the hash issue, I'll do that. I had no idea it was broken, as it worked perfectly fine locally.

As for Vulkan i haven't tried getting it to work yet, but i think it needs some specific driver packages depending on the GPU vendor. Is that something we can include in this package?

Reply:
> At the moment Animal Crossing support is a tad weird, that's not the fault of this. 

I noticed the Yuzu team has some kind of exclusive emulator build for donators, do you know if all the features in these builds originate from github or do they have an entirely separate and private development branch?

> As for the hash issue, I'll do that. Your solution seems to be working perfectly. I had no idea it was broken, as it worked perfectly fine locally.

You wouldn't have any idea, because every time you rebuild using the same package and hash you're just getting the data from your local nix store.

The way I test it is to change a single character of the hash and try to build, it will redownload the data and give you an error with a fresh hash.

> As for Vulkan i haven't tried getting it to work yet, but i think it needs some specific driver packages depending on the GPU vendor. Is that something we can include in this package?

I'm not sure yet, more investigation is needed.

Reply:
> I noticed the Yuzu team has some kind of exclusive emulator build for donators, do you know if all the features in these builds originate from github or do they have an entirely separate and private development branch?

This is their "Early Access" thing, its unfortunately not at all available for Linux. Patreons (their donators) get a token which they can put into the Windows-only installer, which allows them to download these builds from yuzu's private API. To comply with the GPL, this includes the source code which compiles fine on Linux about 60% of the time. Without a Windows VM its essentially not possible to use these builds.

> The way I test it is to change a single character of the hash and try to build, it will redownload the data and give you an error with a fresh hash.

Good to know, I'll keep that in mind.